src/dom-traversal-utils.js

```diff
@@ -0,0 +1,93 @@
+/**
```

Comment:
`text-fragment-utils.js` is actually included by `fragment-generation-utils.js`, so we don't need a new file; we just need to add the functions to `text-fragment-utils.js` and reference them in the `export const internal` dict at the bottom of that file.

I know that file is pretty big and it might make sense to split some of those functions out into a third file, but I'd like to hold off on that for now, because it will affect how we include these files in Chromium.

Reply:
Done, I moved everything to test fragment utils. I was aware of the dependency between fragment generation and text fragment, but I thought it might make sense to put DOM traversal in a separate module, given that it's not directly related to fragment generation or processing. 

What are the consequences of adding more files to the polyfill on the Chromium side?

Reply:
At the very least we'd have to update a few BUILD.gn files and possibly a few (Chromium-side) JS files as well. I don't think it's *super* difficult, but it would mean the next DEPS roll would require extra attention and code-spelunking, so I want to plan around that. 

We also generally try to separate out refactoring changes (behavior-neutral) from bugfixes/features, so we'd want to do that as part of its own pull/DEPS roll.

Also, we might consider extracting the *entire* `internal` namespace into its own file (`util-utils`? 🙃), rather than just the DOM traversal stuff, so that we avoid repeating this step again sometime in the future.

Reply:
> We also generally try to separate out refactoring changes (behavior-neutral) from bugfixes/features, so we'd want to do that as part of its own pull/DEPS roll.

That makes sense. I don't mind splitting this in two PR's: one for moving the traversal methods to TextFragments and one for fixing the bug.

Reply:
> Also, we might consider extracting the _entire_ `internal` namespace into its own file (`util-utils`? 🙃), rather than just the DOM traversal stuff, so that we avoid repeating this step again sometime in the future.

Yes! We can call it shared-utils ( although utils-utils has something to it :) ).